code/game/objects/items/v8_engine.dm

```diff
@@ -0,0 +1,99 @@
+	throw_speed = 1
+	COOLDOWN_DECLARE(engine_sound_cooldown)
+
+/obj/item/v8_engine/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/two_handed, require_twohands=TRUE, force_unwielded=5, force_wielded=5)
+
+/obj/item/v8_engine/attack_self(mob/user, modifiers)
+	. = ..()
+	if (!COOLDOWN_FINISHED(src, engine_sound_cooldown))
+		return
+	playsound(src, 'sound/items/car_engine_start.ogg', vol = 75, vary = FALSE, extrarange = 3)
+	Shake(7, 7, ENGINE_COOLDOWN)
+	to_chat(user, span_notice("Darn thing... it's too old to keep on without retrofitting it! Without modifications, it works like it's junk."))
+	COOLDOWN_START(src, engine_sound_cooldown, ENGINE_COOLDOWN)
+
+/obj/item/v8_engine/examine_more(mob/user)
+	. = ..()
+	var/datum/crafting_recipe/house_edge/edge
+	to_chat(user, span_notice("You peer at the label on the side, reading about some unique modifications that could be made to the engine..."))
+	if(do_after(user, 15 SECONDS, src))
+		if(!user.mind)
+			return
+		user.mind.teach_crafting_recipe(edge)
+		to_chat(user, span_notice("You learned how to make the House Edge."))
+
+/obj/item/house_edge
+	name = "House Edge"
+	desc = "Dangerous. Loud. Sleek. It has a built in roulette wheel. This thing could easily rip your arm off if you're not careful."
+	icon = 'icons/obj/weapons/items_and_weapons.dmi'
+	icon_state = "house_edge0"
+	w_class = WEIGHT_CLASS_HUGE
+	force = 12
+	throwforce = 10
+	throw_range = 5
+	throw_speed = 1
+	hitsound = 'sound/items/car_engine_start.ogg'
+	/// The number of charges the house edge has accrued through 2-handed hits, to charge a more powerful charge attack.
+	var/fire_charges = 0
+	///Sound played when wielded.
+	var/active_hitsound = 'sound/items/house_edge_hit.ogg'
+	///Datum that tracks weapon dashing for the fire_charge system
+	var/datum/action/innate/dash/charge
+	COOLDOWN_DECLARE(fire_charge_cooldown)
+
+/obj/item/house_edge/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/two_handed, force_unwielded = 12, force_wielded = 22, attacksound = active_hitsound)
+	RegisterSignal(src, list(COMSIG_ITEM_DROPPED, COMSIG_MOVABLE_PRE_THROW, COMSIG_ITEM_ATTACK_SELF), .proc/reset_charges)
+
+/obj/item/house_edge/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!ismob(target))
+		return
+	if(HAS_TRAIT(src, TRAIT_WIELDED))
+		//Add a fire charge to a max of 3, updates icon_state.
+		fire_charges = clamp((fire_charges + 1), 0, 3)
+		icon_state = "house_edge[fire_charges]"
+		COOLDOWN_RESET(src, fire_charge_cooldown)
+	else
+		//Lose a fire charge to a min of 0, updates icon_state.
+		fire_charges = clamp((fire_charges - 1), 0, 3)
+		icon_state = "house_edge[fire_charges]"
+		do_sparks(0, TRUE, src)
+
+/obj/item/house_edge/pre_attack_secondary(atom/target, mob/living/user, params)
+	. = ..()
+	if(!COOLDOWN_FINISHED(src, fire_charge_cooldown))
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(fire_charges <= 0)
+		src.balloon_alert(user, "No fire charges")
```

Comment:
is `src` needed here?

Reply:
balloon alerts shouldnt be capitalized, if youre screaming at a player that something is wrong they should use exclamation points